experimental/Schemes/SpecOpen.jl

```diff
@@ -904,18 +966,32 @@ function restrict(f::SpecOpenMor, W::AbsSpec, Y::AbsSpec; check::Bool=true)
 end
 
 
+### the restriction of a morphism to closed subsets in domain and codomain
+function restrict(
+    f::SpecOpenMor,
+    X::SpecType,
+    Y::SpecType;
+    check::Bool=true
+  ) where {SpecType<:Spec}
+  U = intersect(X, domain(f), check=check)
+  V = intersect(Y, codomain(f), check=check)
+
+  new_maps_on_patches = [restrict(f[i], U[i], Y, check=check) for i in 1:npatches(U)]
+
+  return SpecOpenMor(U, V, new_maps_on_patches, check=check)
+end
+
 @Markdown.doc """
-    compose(f::T, g::T) where {T<:SpecOpenMor}
+    compose(f::SpecMor, g::SpecMor) -> SpecMor
```

Comment:
Why did you change `SpecOpenMor` to `SpecMor`? 

Also, we should probably consider removing the documentation for `compose`, since this is merely one method of a much more general function.

Reply:
probably a typo